internal/tofu/evaluate.go

```diff
@@ -440,6 +444,23 @@ func (d *evaluationStateData) GetModule(_ context.Context, addr addrs.ModuleCall
 	// module instance.
 	moduleInstances := map[addrs.InstanceKey]map[string]cty.Value{}
 
+	// First, we need to determine which module instances actually exist.
+	if d.Evaluator.InstanceExpander != nil {
+		childModuleAddr := d.ModulePath.Module().Child(addr.Name)
+		moduleInstanceAddrs := d.Evaluator.InstanceExpander.ExpandModule(childModuleAddr)
+
+		for _, moduleInstanceAddr := range moduleInstanceAddrs {
+			if len(moduleInstanceAddr) > 0 {
+				lastStep := moduleInstanceAddr[len(moduleInstanceAddr)-1]
+				key := lastStep.InstanceKey
+
+				if _, exists := moduleInstances[key]; !exists {
+					moduleInstances[key] = map[string]cty.Value{}
+				}
+			}
```

Comment:
```suggestion
			_, callInstance := moduleInstanceAddr.CallInstance()
			key := callInstance.Key

			if _, exists := moduleInstances[key]; !exists {
				moduleInstances[key] = map[string]cty.Value{}
			}
```

Let's use `CallInstance` as this is the pattern being used in this file.